apps/front/src/app/(auth)/login/loginForm/useLoginForm.ts

```diff
@@ -0,0 +1,63 @@
+import React, { useState } from 'react'
+import { useForm } from 'react-hook-form'
+import { useMutation } from '@apollo/client'
+import { LOGIN_USER } from '@src/graphQL/auth/mutations'
+import { toast } from 'react-toastify'
+import { useRouter } from 'next/navigation'
+import { LoginCredentials } from '@src/types/LoginCredentials'
+import { useDispatch } from 'react-redux'
+import { getClientResponse } from '@src/context/redux/user'
+
+const useLoginForm = () => {
+    const router = useRouter()
+
+    const dispatch = useDispatch()
+
+    const [showPassword, setShowPassword] = useState(false)
```

Comment:
dobrze jest wydzielać logikę tak, żeby w danym hook'u była tylko pojedynczy niezależna logika. Tu widać, że logika od zarządzania wyświetlaniem hasła jest niezależna od obsługi formularza

Reply:
Czy zaproponowałbyś co dokładnie mógłbym z tym zrobić? Jeśli dobrze zrozumiałem to wpadłem na pomysł aby 
`
<InputAdornment position='start'>
                                <IconButton
                                    aria-label='toggle password visibility'
                                    onClick={handleClickShowPassword}
                                    onMouseDown={handleMouseDownPassword}
                                    edge='end'
                                >
                                    {showPassword ? (
                                        <VisibilityOff />
                                    ) : (
                                        <Visibility />
                                    )}
                                </IconButton>
                            </InputAdornment>
`
oraz
` 
const handleClickShowPassword = () => {
        setShowPassword((prevState) => !prevState)
    }

    const handleMouseDownPassword = (
        event: React.MouseEvent<HTMLButtonElement>
    ) => {
        event.preventDefault()
    }
`
Przenieść do nowego shared componentu ponieważ toggle pokazywania hasła używam i na stronie rejestracji i na stronie loginu. Jednakże w moim loginForm i registerForm przekazuję do <FormInput /> 
` type={showPassword ? 'text' : 'password'} ` a więc potrzebuję tego stanu w tych komponentach.